great_expectations/self_check/util.py

```diff
@@ -1317,6 +1317,11 @@ def candidate_test_is_on_temporary_notimplemented_list_v3_api(context, expectati
     return False
 
 
+# We cache whether we can successfully connect to a test backend so we don't have to
+# validate we can connect over and over again.
+_successful_backend_connection = set()
```

Comment:
I haven't plumbed this up to all the connections which means we still do some extra work. I can do this in a followup PR. The redshift connection takes a while to establish so I added it there for now.